runtime-sdk-macros/src/error_derive.rs

```diff
@@ -68,37 +83,96 @@ pub fn derive_error(input: DeriveInput) -> TokenStream {
     gen::wrap_in_const(quote! {
         use #sdk_crate::{
             self as sdk, core::types::Error as RuntimeError, error::Error as _,
+            error::private::AsDynError,
         };
 
         impl sdk::error::Error for #error_ty_ident {
-            fn module_name() -> &'static str {
+            fn module_name(&self) -> &'static str {
                 #module_name
             }
 
             fn code(&self) -> u32 {
                 #code_converter
             }
+
+            fn source(&self) -> Option<&(dyn sdk::error::Error + 'static)> {
+                #source_converter
+            }
         }
 
         impl From<#error_ty_ident> for RuntimeError {
             fn from(err: #error_ty_ident) -> RuntimeError {
-                RuntimeError::new(#error_ty_ident::module_name(), err.code(), &err.to_string())
+                RuntimeError::new(err.module_name(), err.code(), &err.to_string())
             }
         }
     })
 }
 
+fn convert_error_sources(enum_binding: &Ident, variants: &[&ErrorVariant]) -> TokenStream {
+    if variants.is_empty() {
+        return quote!(std::option::Option::None);
```

Comment:
what kind of monster redefines `None`? This line is 100% a good idea, but I shudder to think...

Reply:
He he